benchmarks/Makefile.am

```diff
@@ -253,50 +252,313 @@ benchmarks_protoc_inputs_proto2_message4 =                                 \
+	gogo/cpp_no_group/datasets/google_message4/benchmark_message4_1.pb.cc     \
+	gogo/cpp_no_group/datasets/google_message4/benchmark_message4_2.pb.cc     \
+	gogo/cpp_no_group/datasets/google_message4/benchmark_message4_3.pb.cc
+
+$(cpp_no_group_benchmarks_protoc_outputs): cpp_no_group_protoc_middleman
+$(cpp_no_group_benchmarks_protoc_outputs_header): cpp_no_group_protoc_middleman
+$(cpp_no_group_benchmarks_protoc_outputs_proto2): cpp_no_group_protoc_middleman
+$(cpp_no_group_benchmarks_protoc_outputs_proto2_header): cpp_no_group_protoc_middleman
+
+bin_PROGRAMS += cpp-no-group-benchmark
+cpp_no_group_benchmark_LDADD = $(top_srcdir)/src/libprotobuf.la $(top_srcdir)/third_party/benchmark/src/libbenchmark.a
+cpp_no_group_benchmark_SOURCES = gogo/cpp_no_group/cpp_benchmark.cc
+cpp_no_group_benchmark_CPPFLAGS = -I$(top_srcdir)/src -I$(srcdir)/gogo/cpp_no_group -I$(top_srcdir)/third_party/benchmark/include
+# Explicit deps because BUILT_SOURCES are only done before a "make all/check"
+# so a direct "make test_cpp" could fail if parallel enough.
+# See: https://www.gnu.org/software/automake/manual/html_node/Built-Sources-Example.html#Recording-Dependencies-manually
+cpp_no_group_benchmark-cpp_benchmark.$(OBJEXT): $(cpp_no_group_benchmarks_protoc_outputs) $(cpp_no_group_benchmarks_protoc_outputs_proto2) $(cpp_no_group_benchmarks_protoc_outputs_header) \
+	$(cpp_no_group_benchmarks_protoc_outputs_proto2_header) $(top_srcdir)/third_party/benchmark/src/libbenchmark.a
+nodist_cpp_no_group_benchmark_SOURCES =                                    \
+	$(cpp_no_group_benchmarks_protoc_outputs_proto2)                         \
+	$(cpp_no_group_benchmarks_protoc_outputs)                                \
+	$(cpp_no_group_benchmarks_protoc_outputs_header)                         \
+	$(cpp_no_group_benchmarks_protoc_outputs_proto2_header)
+ 
+gogo_proto_middleman: protoc-gen-gogoproto
+	mkdir -p "tmp/gogo_proto"
+if USE_EXTERNAL_PROTOC
+	$(PROTOC) -I$(srcdir) -I$(top_srcdir) --plugin=protoc-gen-gogoproto --gogoproto_out=./tmp $(benchmarks_protoc_inputs) $(benchmarks_protoc_inputs_benchmark_wrapper) $(benchmarks_protoc_inputs_proto2)
+else
+	oldpwd=`pwd` && ( cd $(srcdir) && $$oldpwd/../src/protoc$(EXEEXT) -I$(srcdir) -I$(top_srcdir) --plugin=protoc-gen-gogoproto --gogoproto_out=$$oldpwd/tmp/gogo_proto $(benchmarks_protoc_inputs) $(benchmarks_protoc_inputs_benchmark_wrapper) $(benchmarks_protoc_inputs_proto2) )
+endif
+	touch gogo_proto_middleman
+
+new_data = $$(for data in $(all_data); do echo "tmp/$$data"; done | xargs)
+
+generate_gogo_data: protoc_middleman protoc_middleman2 gogo-data-scrubber
+	mkdir -p `dirname $(new_data)`
+	./gogo-data-scrubber $(all_data) $(new_data)
+	touch generate_gogo_data
+	
+make_tmp_dir_gogo:
+	mkdir -p tmp/go_no_group
+	mkdir -p tmp/gogofast
+	mkdir -p tmp/gogofaster
+	mkdir -p tmp/gogoslick
+	touch make_tmp_dir_gogo
+
+if USE_EXTERNAL_PROTOC
+
+go_no_group_protoc_middleman: make_tmp_dir_gogo gogo_proto_middleman
```

Comment:
Still need to list all protoc rules for different gogo, since different gogo are different plugins.

Reply:
Any updates on this?

Reply:
Why can't it be parameterized?

Reply:
Do you mean add some variable during configure?